core/src/main/java/org/apache/accumulo/core/iteratorsImpl/system/MapFileIterator.java

```diff
@@ -124,14 +125,15 @@ public Value getTopValue() {
   @Override
   public SortedKeyValueIterator<Key,Value> deepCopy(IteratorEnvironment env) {
     try {
-      @SuppressWarnings("deprecation")
-      SortedKeyValueIterator<Key,Value> other = env.reserveMapFileReader(dirName);
-      ((InterruptibleIterator) other).setInterruptFlag(interruptFlag);
+      Configuration hadoopConf = new Configuration();
+      FileSystem fs = FileSystem.get(hadoopConf);
```

Comment:
Should keep teh config and filesystem objects passed into the constructor instead of creating new ones here.

Reply:
That makes sense. I must've overlooked those. I thought I had checked for them, but I guess I got distracted by the fact that it was using the ones from the environment implementation. I will update that before merging.

Thanks for the thorough review.

Reply:
I made that change, and also made some of the fields final that could be. I noticed that topKey and topValue are never actually assigned... which seems completely incorrect to me. I opened a discussion in Slack about that, to discuss before I try to do anything about that.